code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingCacheUtilities.java

```diff
@@ -0,0 +1,304 @@
+            final byte[] data = new byte[STREAM_WRITE_BUFFER_SIZE];
+            int count;
+
+            while ((count = input.read(data)) != -1) {
+                output.write(data, 0, count);
+            }
+        } catch (final IOException e) {
+            Log.error(LOG_TAG, "%s - IOException while attempting to write remote file (%s)", SELF_TAG, e);
+            return false;
+        } finally {
+            try {
+                if (output != null) {
+                    output.close();
+                }
+            } catch (final IOException e) {
+                Log.error(LOG_TAG, "%s - Unable to close the OutputStream (%s) ", SELF_TAG, e);
+            }
+        }
+
+        return true;
+    }
+
+    // ========================================================================================================
+    // Image asset caching
+    // ========================================================================================================
+
+    /**
+     * Caches the assets provided in the {@link java.util.List}.
+     *
+     * @param assetsCollection a {@link List<String>} containing asset url's to be cached.
+     */
+    void cacheImageAssets(final List<String> assetsCollection) {
+        if (cacheManager == null) {
+            Log.trace(LOG_TAG, "%s - Failed to cache asset, the cache manager is not available.", SELF_TAG);
+            return;
+        }
+
+        final ArrayList<String> assetsToRetain = new ArrayList<>();
+
+        if (assetsCollection != null && !assetsCollection.isEmpty()) {
+            for (final String imageAsset : assetsCollection) {
+                if (assetIsDownloadable(imageAsset) && !assetsToRetain.contains(imageAsset)) {
+                    assetsToRetain.add(imageAsset);
+                }
+            }
+        }
+
+        // clean any cached images which are no longer needed then use the RemoteDownloader to download
+        // any new assets that have not been previously cached.
+        cacheManager.deleteFilesNotInList(assetsToRetain, IMAGES_CACHE_SUBDIRECTORY);
+
+        for (final String asset : assetsToRetain) {
+            try {
+                final RemoteDownloader remoteDownloader = getRemoteDownloader(asset, IMAGES_CACHE_SUBDIRECTORY);
+                remoteDownloader.startDownload();
+            } catch (final MissingPlatformServicesException exception) {
+                Log.warning(LOG_TAG, "%s - Failed to cache asset: %s, the platform services were not available.", SELF_TAG, asset);
+            }
+        }
+    }
+
+    /**
+     * Determine whether the provided {@code String} assetPath is downloadable.
+     *
+     * @param assetPath {@code String} containing the asset path to check
+     * @return {@code boolean} indicating whether the provided asset is downloadable
+     */
+    boolean assetIsDownloadable(final String assetPath) {
+        return StringUtils.stringIsUrl(assetPath) && (assetPath.startsWith("http") || assetPath.startsWith("https"));
```

Comment:
assetPath.startsWith("http") will cover both assetPath.startsWith("http") and assetPath.startsWith("https").